inc/renderer.php

```diff
@@ -622,4 +458,198 @@ function enqueue_styles() {
 			$this->add_inline_css( get_the_ID(), $this->generate_css( get_the_ID() ) );
 		}
 	}
+
+	private function get_panels_data_for_post( $post_id ) {
+		if ( strpos( $post_id, 'prebuilt:' ) === 0 ) {
+			list( $null, $prebuilt_id ) = explode( ':', $post_id, 2 );
+			$layouts     = apply_filters( 'siteorigin_panels_prebuilt_layouts', array() );
+			$panels_data = ! empty( $layouts[ $prebuilt_id ] ) ? $layouts[ $prebuilt_id ] : array();
+		} else if ( $post_id == 'home' ) {
+			$page_id = get_option( 'page_on_front' );
+			if ( empty( $page_id ) ) {
+				$page_id = get_option( 'siteorigin_panels_home_page_id' );
+			}
+
+			$panels_data = ! empty( $page_id ) ? get_post_meta( $page_id, 'panels_data', true ) : null;
+
+			if ( is_null( $panels_data ) ) {
+				// Load the default layout
+				$layouts     = apply_filters( 'siteorigin_panels_prebuilt_layouts', array() );
+				$prebuilt_id = siteorigin_panels_setting( 'home-page-default' ) ? siteorigin_panels_setting( 'home-page-default' ) : 'home';
+
+				$panels_data = ! empty( $layouts[ $prebuilt_id ] ) ? $layouts[ $prebuilt_id ] : current( $layouts );
+			}
+		} else {
+			if ( post_password_required( $post_id ) ) {
+				return false;
+			}
+			$panels_data = get_post_meta( $post_id, 'panels_data', true );
+		}
+
+		return $panels_data;
+	}
+
+	private function migrate_widgets_data( $panels_widgets ) {
```

Comment:
At some point, we should actually look at removing this migration. I think we changed to panels_info before releasing Page Builder 2.0. This doesn't need to happen now, but we should earmark it for removal.

It is performing one step we still need on [line 502](https://github.com/siteorigin/siteorigin-panels/pull/352/files#diff-c80be39f16daedc9fdb681582ecb3432R502).